eth/vm/state.py

```diff
@@ -218,35 +254,42 @@ def account_is_empty(self, address: Address) -> bool:
     #
     # Access self._chaindb
     #
-    def snapshot(self) -> Tuple[bytes, Tuple[UUID, UUID]]:
+    def snapshot(self) -> Tuple[Hash32, UUID, UUID]:
         """
         Perform a full snapshot of the current state.
 
         Snapshots are a combination of the :attr:`~state_root` at the time of the
         snapshot and the id of the changeset from the journaled DB.
         """
-        return (self.state_root, self._account_db.record())
+        return self.state_root, self._account_db.record(), self._storage_db.record()
```

Comment:
The same trick of sharing snapshot ids might be possible here. No problem if you don't want to do it! But if you don't it might be worth opening a ticket for.

Reply:
The issue: https://github.com/ethereum/py-evm/issues/1760